packages/patternfly-4/react-core/src/experimental/components/DataToolbar/DataToolbar.tsx

```diff
@@ -52,6 +59,12 @@ export class DataToolbar extends React.Component<DataToolbarProps, DataToolbarSt
     }));
   }
 
+  updateShowClearFiltersButton = (showClearFiltersButton: boolean) => {
+    if (showClearFiltersButton !== this.state.showClearFiltersButton) {
```

Comment:
You don't actually need to check to see whether a setState is necessary, if you call `this.setState` with the exact same (`===`) value that is already in state, React will optimize that for you and prevent a re-render IIRC. I think this kind of check is only necessary if the value is mutable and React wouldn't be able to tell that a different instance of it represents the same state. But for a boolean, it shouldn't be necessary (I think?)

Reply:
Turns out this might not be true in all cases, I may have misunderstood the API here (just commenting for posterity in case anyone reads this later)